### PR TITLE
fixed redundant graph call

### PR DESCRIPTION
**Pull request by kaykumar**
## Summary
This AADAuthenticationFilter was making a redundant call to Graph , the data was already available in principle 

## Issue Type
- Bug fixing

## Starter Names
  - active directory spring boot starter

## Additional Information

### CODE REVIEW

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAuthenticationFilter.java [hunk 1]
```diff
@@ -81,7 +81,7 @@ protected void doFilterInternal(
                 }
 
                 final Authentication authentication = new PreAuthenticatedAuthenticationToken(
-                            principal, null, client.getGrantedAuthorities(graphApiToken));
+                            principal, null, client.convetGroupToGrantedAuthorities(principal.getUserGroups()));
```
Comment (sophiaso) [inline azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAuthenticationFilter.java]: @kaykumar Thanks for the PR. Typo `convet`? Change to `convertGroupsToGrantedAuthorities`?

File: azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClientTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.springframework.security.core.GrantedAuthority;
+
+import com.microsoft.aad.adal4j.ClientCredential;
+
+import io.jsonwebtoken.lang.Assert;
```
Comment (sophiaso) [inline azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClientTest.java]: Please use `org.junit.Assert`

File: azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClientTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+     @Test
+     public void testConvetGroupToGrantedAuthorities() {
+         
+         final UserGroup userGroup = new UserGroup("testId", "Test_Group");
+         final List<UserGroup> userGroups =  new ArrayList<>();
+         userGroups.add(userGroup);
+         final Set<GrantedAuthority> authorities = adGraphClient.convetGroupToGrantedAuthorities(userGroups);
+         Assert.notEmpty(authorities);
```
Comment (sophiaso) [inline azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClientTest.java]: How about test the mapping authorities should be `ROLE_testId` and `ROLE_Test_Group`? Thank you.

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClient.java [hunk 4]
```diff
@@ -107,6 +107,19 @@ private static String getResponseStringFromConn(HttpURLConnection conn) throws I
...
+        return mappedAuthorities;
+    }
+    
+    
+    /**
+     * Converts UserGroup list to Set of GrantedAutorities
+     * @param groups
+     * @return
+     */
+    public Set<GrantedAuthority> convetGroupToGrantedAuthorities(final List<UserGroup> groups) {
```
Comment (Incarnation-p-lee) [inline azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClient.java]: `Group` to `Groups` or `GroupList` ?

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAuthenticationFilter.java (hunk 1 above)
Comment (kaykumar) [reply azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAuthenticationFilter.java]: @sophiaso I have made this change

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClient.java [hunk 5]
```diff
@@ -107,6 +107,19 @@ private static String getResponseStringFromConn(HttpURLConnection conn) throws I
         // Fetch the authority information from the protected resource using accessToken
         final List<UserGroup> groups = getGroups(graphApiToken);
 
+        // Map the authority information to one or more GrantedAuthority's and add it to mappedAuthorities
+        final Set<GrantedAuthority> mappedAuthorities = convetGroupToGrantedAuthorities(groups);
```
[the remark this answers is not in the document]
Comment (kaykumar) [reply azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClient.java]: @Incarnation-p-lee  I have made this change

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClient.java (hunk 4 above)
Comment (kaykumar) [reply azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClient.java]: fixed

File: azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClientTest.java (hunk 2 above)
Comment (kaykumar) [reply azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClientTest.java]: fixed

File: azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClientTest.java (hunk 3 above)
Comment (kaykumar) [reply azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClientTest.java]: added the test and mocked the required aadTargetGroups

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAuthenticationFilter.java (hunk 1 above)
Comment (sophiaso) [reply azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAuthenticationFilter.java]: Thank you!

**Pull request merged**